engine/classes/Elgg/Views/DirectoryPathRegistry.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Elgg\Views;
+
+use Elgg\Filesystem\Directory;
+use Elgg\Filesystem\File;
+use Elgg\Structs\ArrayCollection;
+use Elgg\Structs\EntryCollectionMap;
+use Elgg\Structs\MapEntry;
+
+/**
+ * A ViewFileRegistry backed by the local filesystem.
+ * 
+ * The directory is expected to be structured like so:
+ *
+ * ```
+ * views/
+ *   $viewtype/
+ *     $view(.php)?
+ * ```
+ * 
+ * For example, this directory exposes the resources/blog/all in two formats:
+ * default and rss.
+ * 
+ * ```
+ * views/
+ *   default/
+ *     resources/
+ *       blog/
+ *         all.php
+ *   rss/
+ *     resources/
+ *       blog/
+ *         all.php
+ * ```
+ * 
+ * @since 2.0.0
+ * @access private
+ */
+class DirectoryPathRegistry implements PathRegistry {
+	/** @var Directory */
+	private $dir;
+	
+	/** @var ViewtypeRegistry */
+	private $viewtypes;
+	
+	/**
+	 * Constructor
+	 * 
+	 * @param Directory        $dir       Filesystem location of the views
+	 * @param ViewtypeRegistry $viewtypes For converting from strings to Viewtype instances
+	 */
+	public function __construct(Directory $dir, ViewtypeRegistry $viewtypes) {
+		$this->dir = $dir;
+		$this->viewtypes = $viewtypes;
+	}
+	
+	/** @inheritDoc */
+	public function getViewtypes() {
+		return $this->dir->getFiles()
```

Comment:
This recursively scans *all* files. Elgg\Filesystem\GaufretteDirectory needs to allow scanning the current dir, not recursive.